components/TradingCardGrid.jsx

```diff
@@ -1,10 +1,22 @@
+import { useFilterContext } from '../context/FilterContext';
 import getFullName from '../util/getFullName';
 import TradingCard from './TradingCard';
 
 export default function TradingCardGrid({ teamMembers }) {
+  const { project, roles, availability } = useFilterContext();
+
+  const filteredTeamMembersByAvailability = teamMembers
+    .filter((member) => ((member.weeklyCapacity - member.forecastedHours) > 0) === availability);
+
+  const filteredTeamMembersByRole = filteredTeamMembersByAvailability
+    .filter((member) => roles.includes(member.jobTitle));
+
+  const filteredTeamMembers = filteredTeamMembersByRole
+    .filter((member) => project === 'all' || member.project === project);
```

Comment:
We ought to be able to combine these into a single function, returning false any time we don't match a condition, then if we make it to the end, we return true:

```suggestion
  const filteredTeamMembers = teamMembers.filter((member) => {
    if (member.weeklyCapacity - member.forcastedHours <= 0) return false;
    
    if (!roles.includes(member.jobTitle)) return false;
    
    if (project !== 'all' && member.project !== project) return false;
    
    return true;
  });
```

Reply:
Would the following still make sense, assuming we keep availability as a boolean and that we are still toggling between available and non-available team members?

```js
  const filteredTeamMembers = teamMembers.filter((member) => {
    if ((member.weeklyCapacity - member.forcastedHours <= 0) === availability) return false;

    if (!roles.includes(member.jobTitle)) return false;

    if (project !== 'all' && member.project !== project) return false;

    return true;
  });
```

Availability is true: positve numbers evaluate to false, availability is true, member included.
Availability is true: 0 evaluates to true, availability is true, member not included.
Availability is false: positive numbers evaluate to false, availability is false, member not included.
Availability is false: 0 evaluates to true, availability is false, member included.

Reply:
That's a clever way to do it, but I pretty much always favor clarity over cleverness. I'm also thinking that using a boolean for `availability` is somewhat limiting. I can see a future where we would want a 'both' option or a 'half-time' option, which would be easier to add if we started with using strings. What if we did something like this?

```js
const hasHoursAvailable = member.weeklyCapacity - member.forecastedHours > 0;
if (!hasHoursAvailable && availability === 'available') return false;

if (hasHoursAvailable && availability === 'unavailable') return false;
```